auslib/test/common/test_rules.py

```diff
@@ -0,0 +1,69 @@
+import json
+from auslib.test.common.base import CommonTestBase
+
+
+class TestPublicRulesAPI(CommonTestBase):
+    def test_get_rules(self):
+        ret = self.public_client.get("/api/v1/rules")
+        got = json.loads(ret.data)
+        self.assertEquals(got["count"], 3)
+
+    def test_get_rules_by_product(self):
+        product = "Fennec"
+        ret = self.public_client.get("/api/v1/rules?product={}".format(product))
+        self.assertEqual(ret.status_code, 200, ret.data)
+        got = json.loads(ret.data)
+        self.assertTrue(got, "Empty dict")
+        self.assertEqual(got["count"], 1)
+        self.assertEqual(got["rules"][0]["product"], product)
+
+    def test_get_rule_by_id(self):
+        ret = self.public_client.get("/api/v1/rules/1")
+        self.assertEqual(ret.status_code, 200, ret.data)
+        got = json.loads(ret.data)
+        self.assertTrue(got, "Empty dict")
+        self.assertEqual(ret.headers['X-Data-Version'], '1')
+
+    def test_get_rule_by_alias(self):
+        expected = dict(priority=90, backgroundRate=100, mapping="b",
+                        update_type="minor", product="Fennec", data_version=1,
+                        alias="moz-releng", buildID=None, buildTarget=None,
+                        channel=None, comment=None, distVersion=None,
+                        distribution=None, fallbackMapping=None,
+                        headerArchitecture=None, locale=None, version=None,
+                        systemCapabilities=None, osVersion=None)
+        ret = self.public_client.get("/api/v1/rules/moz-releng")
+        self.assertEqual(ret.status_code, 200, ret.data)
+        got = json.loads(ret.data)
+        del got["rule_id"]
```

Comment:
This is not the right way to do this :). You should include rule_id in "expected" instead.